src/core_modules/capture-core/components/Pages/EnrollmentEvent/EnrollmentEventPage.container.js

```diff
@@ -18,12 +23,33 @@ export const EnrollmentEventPage = () => {
         }),
         shallowEqual,
     );
-    const [mode] = useState(pageMode.VIEW);
     const { program } = useProgramInfo(programId);
     const programStage = [...program.stages?.values()].find(
         item => item.id === stageId,
     );
+
     return (
-        <EnrollmentEventPageComponent mode={mode} programStage={programStage} />
+        <EnrollmentEventPageComponent
+            mode={showEditEvent ? pageMode.EDIT : pageMode.VIEW}
+            programStage={programStage}
+            onEdit={() => onOpenEditEvent()}
+        />
     );
 };
+
+const mapStateToProps = (state: ReduxState) => ({
+    showEditEvent: state.viewEventPage?.eventDetailsSection?.showEditEvent,
+});
+
+const mapDispatchToProps = (dispatch: ReduxDispatch): any => ({
+    onOpenEditEvent: () => {
+        dispatch(startShowEditEventDataEntry());
+    },
+});
+
+// $FlowSuppress
+// $FlowFixMe[missing-annot] automated comment
+export const EnrollmentEventPage = connect(
+    mapStateToProps,
+    mapDispatchToProps,
+)(EnrollmentEventPagePlain);
```

Comment:
This might be changed / moved based on another comment of mine, but it's great if we use `useDispatch` and `useSelector` instead of the legacy connect.